src/macaron/slsa_analyzer/git_url.py

```diff
@@ -191,67 +175,16 @@ def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "",
     return True
 
 
-def commit_exists(git_obj: Git, digest: str) -> bool:
```

Comment:
The reason I removed this method because it's just a try-except of the `is_ancestor` method. The try-except block is not needed for this command because this method doesn't raise `GitCommandError` (see implementation of that method [here](https://github.com/gitpython-developers/GitPython/blob/7d4f6c68c657586f27125f25b15f02c5a3d7f35c/git/repo/base.py#L699C4-L712C20)). In additions, because this method returns a bool value already, we don't need to cast it.